pilot/pkg/model/validation.go

```diff
@@ -1085,31 +1085,28 @@ func ValidateDestinationRule(msg proto.Message) (errs error) {
 		return fmt.Errorf("cannot cast to destination rule")
 	}
 
-	// TODO: validate short name / FQDN
-	if rule.Name == "" {
-		errs = multierror.Append(errs, fmt.Errorf("destination rule name cannot be empty"))
-	}
-	errs = appendErrors(errs, validateTrafficPolicy(rule.TrafficPolicy))
+	errs = appendErrors(errs,
+		validateHost(rule.Name),
+		validateTrafficPolicy(rule.TrafficPolicy))
+
 	for _, subset := range rule.Subsets {
 		errs = appendErrors(errs, validateSubset(subset))
 	}
 
 	return
 }
 
-//
-func validateTrafficPolicy(policy *routingv2.TrafficPolicy) (errs error) {
+func validateTrafficPolicy(policy *routingv2.TrafficPolicy) error {
 	if policy == nil {
-		return
+		return nil
 	}
 	if policy.OutlierDetection == nil && policy.ConnectionPool == nil && policy.LoadBalancer == nil {
 		return fmt.Errorf("traffic policy must have at least one field")
 	}
 
-	errs = appendErrors(errs, validateOutlierDetection(policy.OutlierDetection))
-	errs = appendErrors(errs, validateConnectionPool(policy.ConnectionPool))
-	errs = appendErrors(errs, validateLoadBalancer(policy.LoadBalancer))
-	return
+	return appendErrors(validateOutlierDetection(policy.OutlierDetection),
+		validateConnectionPool(policy.ConnectionPool),
+		validateLoadBalancer(policy.LoadBalancer))
```

Comment:
Nit: this version kind of "hides" the first validation call that's on the same line as `appendErrors` itself. If we're going to use this style, I prefer it to look like:

```go
return appendErrors(
    validateOutlierDetection(policy.OutlierDetection),
    validateConnectionPool(policy.ConnectionPool),
    validateLoadBalancer(policy.LoadBalancer))
```

Reply:
But no need to fix that in this PR, we can clean it up later.